src/plugins/explore/public/application/utils/state_management/actions/set_dataset/set_dataset.ts

```diff
@@ -40,6 +42,8 @@ export const setDatasetActionCreator = (
   dispatch(clearResults());
   dispatch(clearQueryStatusMap());
   dispatch(clearLastExecutedData());
+  dispatch(setPatternsField(''));
+  dispatch(setUsingRegexPatterns(false));
```

Comment:
if we go with the server side approach then we should be able to avoid potential race condition where if a previous query failed setting this to false while the dataset was updated